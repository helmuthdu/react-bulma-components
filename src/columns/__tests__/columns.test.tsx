import { render } from '@testing-library/react';
import * as React from 'react';
import { Column, Columns } from '..';

describe('Columns component', () => {
  // @ts-ignore
  it.each([[Columns], [Column]])('should render', Component => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have columns one column half width and 3 other as default', () => {
    const { asFragment } = render(
      <Columns>
        <Column size={6}>1</Column>
        <Column>2</Column>
        <Column>3</Column>
        <Column>4</Column>
      </Columns>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render as paragraph', () => {
    const { asFragment } = render(
      <Columns as="div">
        <Column as="p" size={6}>
          1
        </Column>
        <Column as="p">2</Column>
        <Column as="p">3</Column>
        <Column as="p">4</Column>
      </Columns>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have columns one column half width, other narrow and 2 other as default', () => {
    const { asFragment } = render(
      <Columns>
        <Column size={6}>1</Column>
        <Column narrow>2</Column>
        <Column>3</Column>
        <Column>4</Column>
      </Columns>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
