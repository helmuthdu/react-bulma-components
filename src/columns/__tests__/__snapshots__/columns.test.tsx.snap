// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns component should have columns one column half width and 3 other as default 1`] = `
<DocumentFragment>
  <div
    class="columns"
  >
    <div
      class="column is-6"
    >
      1
    </div>
    <div
      class="column"
    >
      2
    </div>
    <div
      class="column"
    >
      3
    </div>
    <div
      class="column"
    >
      4
    </div>
  </div>
</DocumentFragment>
`;

exports[`Columns component should have columns one column half width, other narrow and 2 other as default 1`] = `
<DocumentFragment>
  <div
    class="columns"
  >
    <div
      class="column is-6"
    >
      1
    </div>
    <div
      class="column is-narrow"
    >
      2
    </div>
    <div
      class="column"
    >
      3
    </div>
    <div
      class="column"
    >
      4
    </div>
  </div>
</DocumentFragment>
`;

exports[`Columns component should render 1`] = `
<DocumentFragment>
  <div
    class="columns"
  />
</DocumentFragment>
`;

exports[`Columns component should render 2`] = `
<DocumentFragment>
  <div
    class="column"
  />
</DocumentFragment>
`;

exports[`Columns component should render as paragraph 1`] = `
<DocumentFragment>
  <div
    class="columns"
  >
    <p
      class="column is-6"
    >
      1
    </p>
    <p
      class="column"
    >
      2
    </p>
    <p
      class="column"
    >
      3
    </p>
    <p
      class="column"
    >
      4
    </p>
  </div>
</DocumentFragment>
`;
